File: app/src/main/java/hidden/edu/activity/SplashActivity.java
/*Copyright ©2015 TommyLemon(https://github.com/TommyLemon)

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.*/

package hidden.edu.activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import hidden.edu.R;

import static zuo.biao.library.util.SettingUtil.isSplash;

/**
 * 闪屏activity，保证点击桌面应用图标后无延时响应
 *
 * @author Lemon
 */
public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (!isSplash) {
            Intent intent = new Intent(SplashActivity.this, MainTabActivity.class);
            startActivity(intent);
            finish();
        }
        new Handler().postDelayed(() -> {
            startActivity(MainTabActivity.createIntent(SplashActivity.this));
            finish();
        }, 500);

    }

    @Override
    public void finish() {
        super.finish();
        overridePendingTransition(R.anim.fade, R.anim.hold);
    }
}

File: app/src/main/java/hidden/edu/activity/SplashActivity.kt
package hidden.edu.activity

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import hidden.edu.R
import hidden.edu.activity.MainTabActivity.Companion.createIntent
import qian.xin.library.util.DataKeeper.init
import qian.xin.library.util.SettingUtil

/*
 * 闪屏activity，保证点击桌面应用图标后无延时响应
 */
class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!SettingUtil.isSplash) {
            startActivity(createIntent(this))
            finish()
        } else
            Handler().postDelayed({
                init(this.application)
                setContentView(R.layout.activity_splash)
                startActivity(createIntent(this))
                finish()
            }, 500)
    }

    override fun finish() {
        super.finish()
        overridePendingTransition(R.anim.fade, R.anim.hold)
    }
}